requests_oauthlib/oauth2_session.py

```diff
@@ -64,19 +64,15 @@ def __init__(self, client_id=None, client=None, auto_refresh_url=None,
         :param kwargs: Arguments to pass to the Session constructor.
         """
         super(OAuth2Session, self).__init__(**kwargs)
-        self.client_id = client_id
-        if client is not None and not self.client_id:
-            self.client_id = client.client_id
+        self._client = client or WebApplicationClient(client_id, token=token)
+        self.token = token or {}
         self.scope = scope
         self.redirect_uri = redirect_uri
-        self.token = token or {}
         self.state = state or generate_token
         self._state = state
         self.auto_refresh_url = auto_refresh_url
         self.auto_refresh_kwargs = auto_refresh_kwargs or {}
         self.token_updater = token_updater
-        self._client = client or WebApplicationClient(client_id, token=token)
-        self._client._populate_attributes(token or {})
```

Comment:
Because this function call is in the setter for `self.token`, so it gets called automatically on line 68.